api/graphql/mutations/join_request.js

```diff
@@ -0,0 +1,32 @@
+export async function joinCommunity (communityId, userId) {
+  const user = await User.find(userId)
+  const community = await Community.find(communityId)
+
+  return GroupMembership.forPair(user, community, {includeInactive: true}).fetch()
+    .then(existingMembership => {
+      if (existingMembership) return existingMembership.get('active')
+        ? existingMembership
+        : existingMembership.save({active: true}, {patch: true}).then(membership => membership)
+      if (!!community) return user.joinCommunity(community).then(membership => membership)
```

Comment:
kind of seems like this check for the community should happen sooner in this function? not a big deal but why do anything if you cant find the user or the community?

Reply:
@tibetsprague good point - I pulled this line out and removed the rest - `user.JoinCommunity` leads to `group.addMembers` which covers the case of an existing membership